File: archives/20190519python/0429. N-ary Tree Level Order Traversal.py
"""
# Definition for a Node.
class Node(object):
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""
class Solution(object):
    def levelOrder(self, root):
        """
        :type root: Node
        :rtype: List[List[int]]
        """
        if not root:
            return []
        root_list=[root]
        rep=[]
        while root_list:
            cur_list=[]
            len_root_list=len(root_list)
            for i in range(len_root_list):
                cur_root=root_list.pop(0)
                cur_list.append(cur_root.val)
                for child in cur_root.children:
                    root_list.append(child)
            rep.append(cur_list)
        return rep

File: archives/C#/0429. N-ary Tree Level Order Traversal.cs
/*
// Definition for a Node.
public class Node {
    public int val;
    public IList<Node> children;

    public Node(){}
    public Node(int _val,IList<Node> _children) {
        val = _val;
        children = _children;
}
*/
public class Solution {
    public IList<IList<int>> LevelOrder(Node root) {
        IList<IList<int>> rep=new List<IList<int>>();
        if(root==null){
            return rep;
        }
        Queue<Node> rootList=new Queue<Node>();
        rootList.Enqueue(root);
        
        while(rootList.Count!=0){
            IList<int> tmp=new List<int>();
            int lenOfRootList=rootList.Count;
            for(int i=0;i<lenOfRootList;i++){
                Node curRoot=rootList.Dequeue();
                tmp.Add(curRoot.val);
                foreach(Node node in curRoot.children){
                    rootList.Enqueue(node);
                }
            }
            rep.Add(tmp);
        }
        return rep;
    }
}
